### PR TITLE
[Ready for review] Added CDN Profiles to fluent (Rebased)

### PR DESCRIPTION
**Pull request by hovsepm**
Rebased PR https://github.com/Azure/azure-sdk-for-java/pull/1257 into 2 commits. First one is generated code the second one is Hand crafted.

### CODE REVIEW

**Comment by hovsepm** [comment]
@martinsawicki you can review code by commits now. Please let me know if this is not reviewable as well.

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,937 @@
...
+/**
+ * An immutable client-side representation of an Azure CDN endpoint.
+ */
+@Fluent
+public interface CdnEndpoint extends
+        ExternalChildResource<CdnEndpoint, CdnProfile>,
+        Wrapper<EndpointInner> {
+
+    /**
+     * Gets the origin host header.
```
Comment (martinsawicki) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: minor, but for simple "getters" we do not need a description -- `@return host header` is enough

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/implementation/CdnEndpointImpl.java [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+    public int httpsPort() {
+        if (this.inner().origins() != null && !this.inner().origins().isEmpty()) {
+            return this.inner().origins().get(0).httpsPort();
+        }
+        return 0;
+    }
+
+    @Override
+    public List<String> customDomains() {
+        return Lists.transform(this.customDomainList,
```
Comment (martinsawicki) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/implementation/CdnEndpointImpl.java]: doesn't this return a modifiable list? Might need to go through Collections.unmodifiableList(...). In general, unless they are PagedLists underneath, we return static lists as immutable, which is what unmodifiableList achieves..

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/implementation/CdnEndpointsImpl.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                                        this.parent().resourceGroupName(),
+                                        this.parent().name())) {
+            childResources.add(new CdnEndpointImpl(innerEndpoint.name(),
+                    this.parent(),
+                    innerEndpoint,
+                    this.client,
+                    this.originsClient,
+                    this.customDomainsClient));
+        }
+        return childResources;
```
Comment (martinsawicki) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/implementation/CdnEndpointsImpl.java]: is this list ultimately returned to the user code? If so, shouldn't it go through Collections.unmodifiableList?

File: azure/src/test/java/com/microsoft/azure/management/TestCdn.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                .defineNewEndpoint("somenewnamefortheendpoint")
+                    .withOrigin("www.contoso.com")
+                    .withGeoFilter("/path/music", GeoFilterActions.BLOCK, CountryISOCode.UNITED_STATES_OUTLYING_ISLANDS)
+                    .attach()
+                .updateEndpoint(firstEndpointName)
+                    .withHttpAllowed(true)
+                    .withHttpPort(1111)
+                    .withoutGeoFilters()
+                    .parent()
+                .apply();
```
Comment (martinsawicki) [inline azure/src/test/java/com/microsoft/azure/management/TestCdn.java]: do we have a test of a "without"-er somewhere as well?  (isn't `.withoutEndpoint(name)` possible?)

Comment (hovsepm) [inline azure/src/test/java/com/microsoft/azure/management/TestCdn.java]: it is covered here - https://github.com/Azure/azure-sdk-for-java/pull/1288/commits/13db29f01626527162746437ac6f17b9837d83a6#diff-a2dcc8b4d0226a01115fecc7168730a1R142

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java (hunk 1 above)
Comment (hovsepm) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java]: Fixed.

File: azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/implementation/CdnEndpointsImpl.java (hunk 3 above)
Comment (hovsepm) [inline azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/implementation/CdnEndpointsImpl.java]: Fixed this as well.

**Pull request merged**